lib/core/api.js

```diff
@@ -206,20 +208,31 @@ function postBlockToPlatform(data, groupId, callback) {
       callback(jqxhr.responseJSON, 0);
     }
   };
-
-  tidepool.uploadDeviceDataForUser(data, groupId, function(err, result) {
-    if (err) {
-      return sad(err, err);
-    }
-    return happy(result);
-  });
+  if(!_.isEmpty(groupId)){
+    api.log('postBlockToPlatform: for other user');
+    tidepool.uploadDeviceDataForUser(data, groupId, function(err, result) {
```

Comment:
I'm not sure how I missed this, but I really dislike this pattern where uploadDeviceDataForUser hits a different endpoint depending on whether an argument in the middle of the argument list is present or absent.

I would have handled that in platform-client by leaving uploadDeviceData with no groupId, and uploadDeviceDataForUser with a groupId, and then having them both use a common helper routine.

This JavaScript pattern of testing the type of an argument and then shifting arguments over gives me nightmares. It's a pain to document and a pain to use. 

I'd like to see this fixed but I also understand if that ship has sailed for now.